sellercenter-translation/src/main/java/com/gfg/sellercenter/translation/reader/HttpReader.java

```diff
@@ -1,52 +1,98 @@
+  }
+
+  private String loadResponse(String language, String key) throws IOException, URISyntaxException {
+    URIBuilder urlBuilder = new URIBuilder();
+    URI uri =
+        urlBuilder
+            .setScheme(schema)
+            .setHost(this.domain)
+            .setPath(PATH)
+            .addParameter("language", language)
+            .addParameter("key", key)
+            .build();
 
-    private final String path;
+    try (InputStream stream = uri.toURL().openStream()) {
+      BufferedReader bufferedReader =
+          new BufferedReader(new InputStreamReader(stream, StandardCharsets.UTF_8));
 
-    public JSONObject getTranslation(String language, String key) throws IOException, URISyntaxException {
-        return new JSONObject(this.loadResponse(language, key));
+      return toJsonString(bufferedReader);
     }
+  }
 
-    private String loadResponse(String language, String key) throws IOException, URISyntaxException {
-        URIBuilder urlBuilder = new URIBuilder();
-        URI uri = urlBuilder.setHost(this.fqdnUrl)
-                .setPath(this.path)
-                .addParameter("language", language)
-                .addParameter("key", key)
-                .build();
-
-        try (InputStream stream = uri.toURL().openStream()) {
-            BufferedReader bufferedReader = new BufferedReader(
-                    new InputStreamReader(stream, StandardCharsets.UTF_8)
-            );
-
-            return toJsonString(bufferedReader);
-        }
+  public JSONObject getTranslations(JSONObject payloadTranslationRequest)
+      throws IOException, URISyntaxException {
+
+    URIBuilder urlBuilder = new URIBuilder();
+    URI uri = urlBuilder.setScheme(this.schema).setHost(this.domain).setPath(PATH).build();
+
+    HttpPost postRequest = new HttpPost(uri.toURL().toString());
+
+    StringEntity entity = new StringEntity(payloadTranslationRequest.toString());
+    postRequest.setEntity(entity);
+    postRequest.setHeader("Accept", "application/json");
+    postRequest.setHeader("Content-type", "application/json");
+
+    CloseableHttpResponse response = this.getHttpClient().execute(postRequest);
+
+    try (InputStream stream = response.getEntity().getContent()) {
+      BufferedReader bufferedReader =
+          new BufferedReader(new InputStreamReader(stream, StandardCharsets.UTF_8));
+      this.getHttpClient().close();
+      return new JSONObject(toJsonString(bufferedReader));
     }
+  }
 
-    private String toJsonString(BufferedReader reader) throws IOException {
-        StringBuilder sb = new StringBuilder();
-        String line = reader.readLine();
-        while (line != null) {
-            sb.append(line);
-            line = reader.readLine();
-        }
-        return sb.toString();
+  private String toJsonString(BufferedReader reader) throws IOException {
+    StringBuilder sb = new StringBuilder();
+    String line = reader.readLine();
+    while (line != null) {
```

Comment:
it make sense to check  if isEmpty?

Reply:
Empty is some time expected ?! ;) Even so, this code of line is used in all java Packages.